src/bin.ts

```diff
@@ -18,4 +22,7 @@ await awaken({ url: initUrl, onAwaken, results });
 const end = Date.now();
 const seconds = (end - start) / 1000;
 console.log(`Done! Awakened ${results.size} links in ${seconds} seconds`);
-process.exit(0); // TODO: should this exit with 1 if there were errors?
\ No newline at end of file
+
+if (!isSuccessful) {
+  process.exitCode = 1;
+}
```

Comment:
```suggestion
process.exit(isSuccessful ? 0 : 1);
```

I had to use process.exit(0) to make sure the `fetch` didn't hang since it seems to keep the connections open even after all responses are complete